database/3ds/get_friend_persistent_infos.go

```diff
@@ -41,9 +35,9 @@ func GetFriendPersistentInfos(user1_pid uint32, pids []uint32) []*friends_3ds_ty
 			SELECT date FROM "3ds".friendships WHERE user1_pid=$1 AND user2_pid=$2 AND type=0 LIMIT 1`, user1_pid, persistentInfo.PID).Scan(&friendedAtTime)
 		if err != nil {
 			if err == sql.ErrNoRows {
-				friendedAtTime = uint64(time.Now().Unix())
+				return make([]*friends_3ds_types.FriendPersistentInfo, 0), database.ErrFriendshipNotFound
```

Comment:
Since this returns an error, the first return value can just be `nil`. Returning an initilized slice like this can give the impression that the return value can still be used, when it should not. It's also _slightly_ more performant since not calling `make` won't run all the initialization

This happens elsewhere